graphql-dgs/src/main/java/com/netflix/graphql/dgs/DgsCodeRegistryBuilder.java

```diff
@@ -0,0 +1,75 @@
+/*
+ * Copyright 2024 Netflix, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.netflix.graphql.dgs;
+
+import com.netflix.graphql.dgs.internal.DataFetcherResultProcessor;
+import graphql.schema.DataFetcher;
+import graphql.schema.DataFetcherFactories;
+import graphql.schema.FieldCoordinates;
+import graphql.schema.GraphQLCodeRegistry;
+import graphql.schema.GraphQLFieldDefinition;
+
+import java.util.List;
+
+/**
+ * Utility wrapper for {@link GraphQLCodeRegistry.Builder} which provides
+ * a consistent registration mechanism of DataFetchers similar to the annotation-based approach.
+ * Can be used as a first parameter of a {@link DgsCodeRegistry} annotated method.
+ */
+public class DgsCodeRegistryBuilder {
```

Comment:
Is there a reason this class is written in Java instead of Kotlin?

Reply:
Because the other classes in this package are also in Java. I suppose this package is different from the "internal" package where there is Kotlin

Reply:
That's not really the case, the things in Java are mostly interfaces and annotations, and are there due to compatibility issues with older versions of Kotlin.

Reply:
Interesting. I will look at how to rewrite in Kotlin then. Thanks